File: Handin2-benchmark/data_generator.py
import random
import math


def generate_points_square(amount):
    points = []

    while len(points) < amount:
        x = random.random() * 100
        y = random.random() * 100

        points.append((x, y))
    return points


def generate_points_circle(amount):
    points = []

    while len(points) < amount:
        r = math.sqrt(random.random()) * 100
        theta = random.random() * 2 * math.pi
        x = r * math.cos(theta)
        y = r * math.sin(theta)

        points.append((x, y))
    return points


def generate_points_curve(amount):

    used_x = set()
    points = []

    while len(points) < amount:
        x = random.randrange(0, 1000000) / 1000.0
        y = x*x

        if x in used_x:
            continue
        else:
            used_x.add(x)
            points.append((x, y))
    return points

File: Handin2/src/app/data_generator.rs
use std::f64::consts::PI;
use crate::Point;
use rand::prelude::*;
use rand::rngs::StdRng;

pub fn generate_points_square(amount: usize, seed: u64) -> Vec<Point> {
    let mut rng = StdRng::seed_from_u64(seed);
    let mut points = Vec::<Point>::new();

    loop {

        let x = rng.gen::<f64>() * 100.0;
        let y = rng.gen::<f64>() * 100.0;

        if points.iter().any(|&p| p.x == x || p.y == y) {
            continue
        }
        points.push(Point {x, y});

        if points.len() >= amount {
            return points;
        }
    }
}

pub fn generate_points_circle(amount: usize, seed: u64) -> Vec<Point> {
    let mut rng = StdRng::seed_from_u64(seed);
    let mut points = Vec::<Point>::new();

    loop {

        let r = 100.0 * rng.gen::<f64>().sqrt();
        let theta = rng.gen::<f64>() * 2.0 * PI;
        let x = r * theta.cos();
        let y = r * theta.sin();

        if points.iter().any(|&p| p.x == x || p.y == y) {
            continue
        }
        points.push(Point {x, y});

        if points.len() >= amount {
            return points;
        }
    }
}

pub fn generate_points_curve(amount: usize, seed: u64) -> Vec<Point> {
    let mut rng = StdRng::seed_from_u64(seed);
    let mut points = Vec::<Point>::new();

    loop {
        // Lower precision for x, so that there is enough precision space for y
        let x = (((rng.gen::<f32>() * 1000000.0) as i64) as f64) / 1000.0;
        let y = x*x;

        if points.iter().any(|&p| p.x == x || p.y == y) {
            continue
        }
        points.push(Point {x, y});

        if points.len() >= amount {
            return points;
        }
    }
}
